service/header_service.go

```diff
@@ -291,6 +292,66 @@ func (hs *HeaderService) GetMerkleRootsConfirmations(
 	return hs.repo.Headers.GetMerkleRootsConfirmations(request, hs.merkleCfg.MaxBlockHeightExcess)
 }
 
+// LocateHeadersGetHeaders returns headers with given hashes.
+func (hs *HeaderService) LocateHeadersGetHeaders(locators []*chainhash.Hash, hashstop *chainhash.Hash) []*wire.BlockHeader {
+	headers := hs.locateHeadersGetHeaders(locators, hashstop)
+	return headers
+}
+
+func (hs *HeaderService) locateHeadersGetHeaders(locators []*chainhash.Hash, hashstop *chainhash.Hash) []*wire.BlockHeader {
+	hashes := make([]string, len(locators))
+	for i, v := range locators {
+		hashes[i] = v.String()
+	}
+
+	startHeight, err := hs.repo.Headers.GetHeadersStartHeight(hashes)
+	if err != nil {
+		log.Trace().Msgf("Error getting headers of locators: %v", err)
+		return nil
+	}
```

Comment:
Hmmm... maybe it's good idea actually to return errors from HeadersService, and log.Error at the Peer side, so it could add some context of the peer to the message (like IP, Client Agent, etc.)

Reply:
changed to handle errors log at the peer side